src/cmd/linuxkit/moby/docker.go

```diff
@@ -13,12 +14,23 @@ import (
 	"strings"
 
 	"github.com/containerd/containerd/reference"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	distref "github.com/docker/distribution/reference"
```

Comment:
AFAICT this is needed because the "github.com/containerd/containerd/reference" one above insists on the `docker.io` prefix being included, which we don't want to do in this context. That's a shame but I can't see a way to make this work with the containerd one as is.

Reply:
indeed. It's all a bit of a mess...

FWIW, the code which does `docker.io` -> `https://index...` is here: https://github.com/moby/moby/blob/master/registry/config.go#L405

Instead of faffing about with yet another type I replicated it below...which seems quite common